src/axom/sidre/core/Group.hpp

```diff
@@ -896,14 +896,34 @@ class Group
    * the View is not copied. The new View object is associated with
    * the same data as the original.
    *
-   * If given Group pointer is null or Group already has a child Group with
-   * same name as given Group, method is a no-op.
+   * If given View pointer is null or Group already has a View with
+   * same name as given View, method is a no-op.
    *
-   * \return pointer to given argument Group object or nullptr if Group
-   * is not moved into this Group.
+   * \return pointer to the new copied View object or nullptr if a View
+   * is not successfully copied into this Group.
    */
   View* copyView(View* view);
 
+  /*!
+   * \brief Create a deep copy of given View object and add it to this Group.
+   *
+   * Note that for Views that use sidre Buffers or external data, the
+   * deep copy performs a copy of the data described by the View
+   * into a new Buffer attached to the destination View. If the source View
+   * describes only part of a Buffer or part of an external array, due to
+   * strides and or offsets into the data, only the values seen by the 
+   * description will be copied into the new View. The new View will have
+   * a Buffer that is the exact size of the number of values that are copied,
+   * with an offset of zero and a stride of one.
```

Comment:
I think we'd need to define what we want for that case.  If we deep copy a strided View and give it a new Buffer with the same strides, then that Buffer will have the correct copied values in the strided locations as described by the View, and undefined data in the parts of the Buffer outside the View's description.  Is that something we want to do for the deep copy of a single View?

If your example is a single Buffer with three strided Views `x, y, z` attached to the Buffer, you may desire a use case where you deep copy the full Buffer and then create three new Views to attach to the new Buffer, but that isn't what I was attempting to implement here.  Based on previous discussions, the decision was to create a new Buffer for each deep copy of a View.

The tools to do that kind of single Buffer, multiple View copy are available now, but you'd have to do it step-by-step as there isn't a convenient single-call method for it.

Reply:
Keeping strides and offsets for the general case can be very inefficient. 

For example, some codes allocate a single buffer, with many views that index into for all of their mesh and field data. 

If you try to deep copy a subgroup (say one field) and preserve the original striding, instead of copying just the data described by views in the group -- you will copy everything in the buffer (double mem usage) and most of it would be unused. 

For the specific case where you know you need an interleaved dataset, you may want a helper that gives you copies with those guarantees.

Reply:
I agree that this is an "advanced" use-case and does not need to be directly supported by Sidre. We might consider adding a similar case if/when we create a Sidre "cookbook".

In the meantime, it would probably suffice to just note this in the `deepCopyView` and `deepCopyGroup` functions.

Reply:
@nselliott (and others) -- would it make sense to add a function to `sidre::Buffer` to copy the data from a buffer to a new buffer and return a pointer to the new buffer -- e.g. `Buffer::copyBuffer() -> Buffer*` ?

On the other hand, perhaps this is intentional since the copy that I was proposing would be for the data but not its attached views?

Reply:
@kennyweiss We don't have that in Buffer because all Buffers must be owned by the DataStore, so a `DataStore::copyBuffer()` would be more appropriate.  This is another thing you could do now with two or three lines of code, and maybe a method would be useful, but it would have no Views.  You could do it and then manually attach Views according to the specific end result you desire.